tss-esapi/src/structures/pcr/selection.rs

```diff
@@ -144,31 +143,35 @@ impl PcrSelection {
 
 impl TryFrom<TPMS_PCR_SELECTION> for PcrSelection {
     type Error = Error;
+
     fn try_from(tss_pcr_selection: TPMS_PCR_SELECTION) -> Result<Self> {
-        Ok(PcrSelection {
-            // Parse hashing algorithm.
-            hashing_algorithm: HashingAlgorithm::try_from(tss_pcr_selection.hash).map_err(|e| {
+        // Parse hashing algorithm.
+        let hashing_algorithm =
+            HashingAlgorithm::try_from(tss_pcr_selection.hash).map_err(|e| {
                 error!("Error converting hash to a HashingAlgorithm: {}", e);
                 Error::local_error(WrapperErrorKind::InvalidParam)
-            })?,
-            // Parse the sizeofSelect into a SelectSize.
-            size_of_select: PcrSelectSize::from_u8(tss_pcr_selection.sizeofSelect).ok_or_else(
-                || {
-                    error!(
-                        "Error converting sizeofSelect to a SelectSize: Invalid value {}",
-                        tss_pcr_selection.sizeofSelect
-                    );
-                    Error::local_error(WrapperErrorKind::InvalidParam)
-                },
-            )?,
-            // Parse selected pcrs into BitFlags
-            selected_pcrs: BitFlags::<PcrSlot>::try_from(u32::from_le_bytes(
-                tss_pcr_selection.pcrSelect,
-            ))
+            })?;
+
+        // Parse the sizeofSelect into a SelectSize.
+        let size_of_select = PcrSelectSize::try_from(tss_pcr_selection.sizeofSelect)?;
+
+        // Select only the octets indicated by sizeofSelect
+        let mut selected_octets = [0u8; TPM2_PCR_SELECT_MAX as usize];
+        let number_of_selected_octets: usize = size_of_select.into();
+        selected_octets[..number_of_selected_octets]
+            .copy_from_slice(&tss_pcr_selection.pcrSelect[..number_of_selected_octets]);
```

Comment:
The specification says the following:
>TPM. The minimum value for sizeofSelect is: 
>PCR_SELECT_MIN ≔ (PLATFORM_PCR + 7) / 8  (1) 
>where 
>PLATFORM_PCR  the number of PCR required by the platform-specific specification

So no there is no hard lower limit. Though I guess it needs to be greater then 0.

Reply:
Though when I asked the people at tpm2-tss about the different select-like structures they said that TPMS_PCR_SELECT was checked to have a size of select of at least three octets. Maybe because all the platforms they support has that limit.

Reply:
Will be covered by the discussion below as well.